src/gui/guiutility.cpp

```diff
@@ -104,6 +104,15 @@ bool Utility::internetConnectionIsMetered()
     return false;
 }
 
+bool OCC::Utility::internetThroughCaptivePortal()
```

Comment:
I'm wondering whether the null pointer check justifies the overhead of moving the code to utils.
In the end it hides the use of QNetworkInformation and can make reading the code in the future more complicated (also the utils class is already quire crowded)
This was also done for `internetConnectionIsMetered`.